src/account/UpgradeableModularAccount.sol

```diff
@@ -262,6 +262,41 @@ contract UpgradeableModularAccount is
         return returnData;
     }
 
+    /// @inheritdoc IPluginExecutor
+    function executeWithAuthorization(bytes calldata data, bytes calldata authorization)
+        external
+        payable
+        returns (bytes memory)
+    {
+        bytes4 execSelector = bytes4(data[0:4]);
+
+        FunctionReference runtimeValidationFunction = FunctionReference.wrap(bytes21(authorization[0:21]));
```

Comment:
```suggestion
        FunctionReference runtimeValidationFunction = FunctionReference.wrap(bytes21(authorization));
```
This is safe to do, but omits some explicitness in exchange for a little bit of gas savings (even with via-ir), but not a hard requirement on my end.

Although technically speaking, it'd be very weird if solidity were to change bytesN(bytes) casting down the line, it is possible.

Reply:
Ah, I should probably document this in a comment, but the intention here was to explicitly revert when length < 21, rather than to pad it with zeroes. The default behavior when casting a `bytes` type to a `bytesN` type, if the `bytes` type is shorter, is to right-pad the returned `bytesN` type. Arguably that may be fine here, since there is no further calldata, but to keep things simple for now I wanted it to revert if there isn't enough data.

Reply:
Ahh makes sense! In that case, best keep it as is. Resolving!